source/common/http/utility.cc

```diff
@@ -581,8 +581,10 @@ void Utility::sendLocalReply(const bool& is_reset, const EncodeFunctions& encode
   ResponseHeaderMapPtr response_headers{createHeaderMap<ResponseHeaderMapImpl>(
       {{Headers::get().Status, std::to_string(enumToInt(response_code))}})};
 
+  bool has_custom_content_type = false;
   if (encode_functions.modify_headers_) {
     encode_functions.modify_headers_(*response_headers);
+    has_custom_content_type = response_headers->ContentType() == nullptr;
```

Comment:
This part is hard to understand, "modify_headers_" function is used to modify response headers from local_reply_config.  If it has "content_type" headers to add, response_headers should have content type,  so it should be"

```
has_custom_content_type = (response_headers->ContentType() != nullptr; )

Reply:
oh yeah this was draft because it wasn't ready for review :-)  I'd just had 11 unrelated tests failing locally and wanted to make sure it was my machine, not local reply changes :-)